client/src/components/Ticketing/ticketingmanager/showings/ShowingUpdated/EventGeneralForm.tsx

```diff
@@ -0,0 +1,171 @@
+import React, {useState} from 'react';
+import {Field, Formik} from 'formik';
+import {InputControl} from './InputControl';
+import {eventGeneralSchema} from './event.schemas';
+import {FormDeleteButton} from './FormDeleteButton';
+import {FormSubmitButton} from './FormSubmitButton';
+import {useEvent} from './EventProvider';
+import {EventImage} from '../../../../../utils/imageURLValidation';
+
+interface EventGeneralFormProps {
+  onSubmit: (event, actions) => void;
+  onDelete: (event) => void;
+  onLeaveEdit?: () => void;
+}
+export const EventGeneralForm = (props: EventGeneralFormProps) => {
+  const {onSubmit, onDelete, onLeaveEdit} = props;
+  const {eventData} = useEvent();
+  const [disabledURL, setDisabledURL] = useState(
+    eventData?.imageurl === 'Default Event Image',
+  );
+
+  const baseValues = {
+    eventname: eventData ? eventData.eventname : '',
+    eventid: eventData ? eventData.eventid : 0,
+    eventdescription: eventData ? eventData.eventdescription : '',
+    imageurl: eventData ? eventData.imageurl : '',
+    active: eventData ? eventData.active : true,
+    seasonid_fk: eventData ? eventData.seasonid_fk : 7,
+  };
+
+  return (
+
+    <Formik
+      initialValues={baseValues}
+      onSubmit={onSubmit}
+      validationSchema={eventGeneralSchema}
+    >
+      {({handleSubmit, values, setFieldValue}) => (
+        <form className={'bg-white flex flex-col  p-6 rounded-xl shadow-xl'} onSubmit={handleSubmit}>
+          <div className={'grid grid-cols-12 mb-5 gap-2'}>
+            <h2 className={'col-span-12 min-[650px]:col-span-6 text-center min-[650px]:text-start text-3xl font-semibold text-zinc-800'}>
+              {eventData? 'Edit Event': 'Add Event'}
+            </h2>
+            <div className={'col-span-12 min-[650px]:col-span-6 flex flex-row gap-4 flex-wrap justify-center min-[650px]:justify-end'}>
+              <FormSubmitButton />
+              {eventData && (
+                <FormDeleteButton
+                  onDelete={onDelete}
```

Comment:
When deleting a full event, it would be a good idea to have an "are you sure?" modal just for safety. It can exactly mirror the "Are you sure?" modal from the cart. Ideally, use that exact component (or modify it) instead of copy pasting it

![image](https://github.com/WonderTix/WonderTix/assets/92124260/e92f7694-4b7f-4ede-afd0-a115e6ace7ef)